### PR TITLE
fix dirty is not added in docker image tag

### PR DESCRIPTION
**Pull request by ikaven1024**
Signed-off-by: yingjinhui <yingjinhui@didiglobal.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
```bash
$ echo >> README.md

$ make image-karmada-controller-manager
Building image for linux/amd64: docker.io/karmada/karmada-controller-manager:v1.3.0-452-g63a67d7c
...
```

Expect image: `docker.io/karmada/karmada-controller-manager:v1.3.0-452-g63a67d7c-dirty`

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
fix dirty is not added in docker image tag
```

### CODE REVIEW

**Review by RainbowMango — COMMENTED** [review]
LGTM.

@gy95 Could you please take a look?

File: Makefile [hunk 1]
```diff
@@ -1,25 +1,13 @@
...
 # Images management
 REGISTRY?="docker.io/karmada"
 REGISTRY_USER_NAME?=""
 REGISTRY_PASSWORD?=""
 REGISTRY_SERVER_ADDRESS?=""
 
-# Set your version by env or using latest tags from git
-VERSION?=""
-ifeq ($(VERSION), "")
-    LATEST_TAG=$(shell git describe --tags)
```
Comment (gy95) [inline Makefile:14]: binary version and image tag should be kept the same.
For image tag, it comes here.
we just need to add a `--dirty` flag `LATEST_TAG=$(shell git describe --tags --dirty)`?

Comment (ikaven1024) [reply Makefile:14]: As you say `binary version and image tag should be kept the same.`  It's best way to use the same code. So I refactor it to get from `util::get_version` function.

If you keep the two ways to get version. It's easy to make different results, just like this bug.

File: hack/build.sh [hunk 2]
```diff
@@ -23,6 +23,8 @@ set -o pipefail
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 source "${REPO_ROOT}/hack/util.sh"
 
+LDFLAGS="$(util::version_ldflags) ${LDFLAGS:-}"
```
Comment (gy95) [inline hack/build.sh:26]: this is confusing

Comment (gy95) [reply hack/build.sh:26]: LDFLAGS="${LDFLAGS}:-$(util::version_ldflags)}" ? should be like this?

Comment (ikaven1024) [reply hack/build.sh:26]: It's not a mistake. Users can set custom `ldflags` by `LDFLAGS` variable., not only version info. FYI:
https://github.com/kubernetes/kubernetes/blob/a2408f39c7233fc7ceb06795ffd24b84ca471243/hack/lib/golang.sh#L847

Comment (gy95) [reply hack/build.sh:26]: /lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**